### PR TITLE
[SW-2276] The .getAlgo() Method of Pysparkling H2OGridSearch Throws Exception

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: py/src/ai/h2o/sparkling/ml/algos/H2OGridSearch.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def __init__(self,
         self._setDefaultValuesFromJava()
         kwargs = Utils.getInputKwargs(self)
         self._set(**kwargs)
+        self._transfer_params_to_java()
```
Comment (jakubhava) [inline py/src/ai/h2o/sparkling/ml/algos/H2OGridSearch.py:64]: Required so the algo passed in constructor is transferred to java

**Comment by mn-mikke** [comment]
it seems that there is still problem if algorithm is not set via constructor:
```
>>> from pysparkling.ml import *
>>> grid = H2OGridSearch(labelCol="AGE", hyperParameters={"seed": [1, 2, 3]}, splitRatio=0.8, strategy="RandomDiscrete", maxModels=3, maxRuntimeSecs=60, selectBestModelBy="RMSE")
>>> grid.setAlgo(H2OGLM())
H2OGridSearch_29f879dce4ce
>>> grid.getAlgo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/var/folders/90/mnytclln4yxcykll81knhmzm0000gn/T/spark-8adffbc5-0688-4163-a7fb-17c983a1233a/userFiles-9f60e49f-c463-47e6-8b43-28e4c1c2a9fe/h2o_pysparkling_2.4-3.32.0.1-1-2.4.zip/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py", line 95, in getAlgo
AttributeError: 'NoneType' object has no attribute 'parameters'
```

**Comment by jakubhava** [comment]
Have you tried on the last version of this PR? I think I fixed this in the second commit

**Comment by mn-mikke** [comment]
I did pull before that, but will try once again.

**Comment by mn-mikke** [comment]
It works on my side, the pull before failed on conflicts with local changes. Thanks!

**Comment by jakubhava** [comment]
perfect!

File: py/tests/unit/with_runtime_sparkling/test_gridsearch.py [hunk 2]
```diff
@@ -104,3 +104,21 @@ def testGetGridModels(prostateDataset):
     grid.fit(prostateDataset)
     models = grid.getGridModels()
     assert len(models) == 3
+
```
[the remark this answers is not in the document]
Comment (jakubhava) [reply py/tests/unit/with_runtime_sparkling/test_gridsearch.py:107]: thanks!

**Comment by jakubhava** [comment]
Verified by running local tests and merging :)

**Pull request merged**